inspirehep/modules/disambiguation/receivers.py

```diff
@@ -0,0 +1,136 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of INSPIRE.
+# Copyright (C) 2016 CERN.
+#
+# INSPIRE is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# INSPIRE is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with INSPIRE. If not, see <http://www.gnu.org/licenses/>.
+#
+# In applying this licence, CERN does not waive the privileges and immunities
+# granted to it by virtue of its status as an Intergovernmental Organization
+# or submit itself to any jurisdiction.
+
+"""Receivers to catch new publications and record updates."""
+
+from __future__ import (
+    absolute_import,
+    division,
+    print_function,
+)
+
+from invenio_indexer.signals import before_record_index
+from invenio_records.signals import after_record_insert
+
+from inspirehep.modules.disambiguation.models import DisambiguationRecord
+
+
+def _needs_beard_reprocessing(authors_before, authors_after):
+    """Check if the update affected authors or affiliation(s).
+
+    The method receives a list of authors of a record before
+    and after the update was made. Then it compares full names
+    of the authors together with affiliations to decide
+    if the record should be processed by Beard.
+
+    The Beard model is trained against full names and affiliations
+    If any of these fields were changes, then the HEP paper should
+    be processed.
+
+    :param authors_before:
+        A list of authors of a particular HEP paper.
+
+        Example:
+            authors_before = [{
+                u'full_name': u'Mukherjee, Anirbit',
+                u'uuid': u'76f5e291-b709-4aa9-ae7c-99f361bae591'
+            }]
+
+    :param authors_before:
+        A list of authors of a particular HEP paper.
+
+        Example:
+            authors_after = [{
+                u'full_name': u'Mukherjee, Anirbit',
+                u'affiliations': [{'value': 'Perimeter Inst. Theor. Phys.'}],
+                u'uuid': u'76f5e291-b709-4aa9-ae7c-99f361bae591'
+            }]
+
+    :return:
+        Boolean value if the HEP paper should be processed by Beard or not.
+
+        Example:
+            True
+    """
+    if len(authors_before) == len(authors_after):
+        for index, author_before in enumerate(authors_before):
+            # Not every author has an affiliation.
+            before_affiliations = author_before.get(
+                'affiliations', [])
+
+            # We don't iterate over authors_after, we take the index.
+            after_affiliations = authors_after[index].get(
+                'affiliations', [])
```

Comment:
FYI, this should have been a [`zip`](https://docs.python.org/2/library/functions.html#zip), which is the pythonic way of iterating on two lists in parallel.